demo/cmd/bookwarehouse/bookwarehouse.go

```diff
@@ -49,10 +49,21 @@ func getBooksStockedRecord() Record {
 	return record
 }
 
-// restockBooks decreases the balance of the given bookwarehouse account.
-func restockBooks(w http.ResponseWriter, r *http.Request) {
+func setHeaders(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set(common.IdentityHeader, getIdentity())
 
+	if r != nil {
```

Comment:
You can remove a nested layer:

```suggestion
	if r == nil {
	    return
	}
```